pandas/core/indexes/datetimelike.py

```diff
@@ -91,14 +92,51 @@ def _create_comparison_method(cls, op):
         Create a comparison method that dispatches to ``cls.values``.
         """
         def wrapper(self, other):
-            result = op(self._eadata, maybe_unwrap_index(other))
+            result = op(self._data, maybe_unwrap_index(other))
             return result
 
         wrapper.__doc__ = op.__doc__
         wrapper.__name__ = '__{}__'.format(op.__name__)
         return wrapper
 
     # ------------------------------------------------------------------------
+    # Abstract data attributes
+
+    @property
+    def _values(self):
+        return self._data
+
+    @property
+    def values(self):
+        # type: () -> np.ndarray
+        # Note: PeriodArray overrides this to return an ndarray of objects.
+        return self._data._data
+
+    @property
+    @Appender(DatetimeLikeArrayMixin.asi8.__doc__)
+    def asi8(self):
+        return self._data.asi8
+
+    # ------------------------------------------------------------------------
+
+    # Note: moved from DatetimeLikeArrayMixin
+    @property
+    def offset(self):
```

Comment:
this is currently in DatetimeIndex, will need to be removed from there

Reply:
This should be removed from here right? 0.23.4 doesn't have an offset for PeriodIndex or TimedeltaIndex.

Reply:
Oh ok.  Then I should also revert this in #24517.